ci_scripts/run_tests.sh

```diff
@@ -1,10 +1,98 @@
+function die {
+  echo "[$(basename "${0}")] [ERROR] ${1}"
+  exit 1
+}
+
+# Verify xcpretty is installed
+if ! command -v xcpretty > /dev/null; then
+  if [[ "${CI}" != "true" ]]; then
+    die "Please install xcpretty: https://github.com/supermarin/xcpretty#installation"
+  fi
+
+  info "Installing xcpretty..."
+  gem install xcpretty --no-ri --no-rdoc || die "Executing \`gem install xcpretty\` failed"
+fi
+
+# Install test dependencies
+info "Installing test dependencies..."
+
+carthage bootstrap --platform iOS --configuration Release --no-use-binaries
+carthage_exit_code="$?"
+
+if [[ "${carthage_exit_code}" != 0 ]]; then
+  die "Executing carthage failed with status code: ${carthage_exit_code}"
+fi
+
+# Execute tests (iPhone 6 @ iOS 11.2)
+info "Executing tests (iPhone 6 @ iOS 11.2)..."
+
+xcodebuild clean test \
+  -workspace "Stripe.xcworkspace" \
+  -scheme "StripeiOS" \
+  -configuration "Debug" \
+  -sdk "iphonesimulator" \
+  -destination "platform=iOS Simulator,name=iPhone 6,OS=11.2" \
+  | xcpretty
+
+exit_code="${PIPESTATUS[0]}"
+
+if [[ "${exit_code}" != 0 ]]; then
+  die "xcodebuild exited with non-zero status code: ${exit_code}"
+fi
+
+# Execute tests on legacy devices (iPhone 6 @ iOS 10.x, iPhone 6 @ iOS 9.x)
+# - Skips snapshot tests because they're recorded for the iPhone 6 on the newest iOS version only
+# - Not sure why tests STPImageLibraryTest fail on older iOS versions
+info "Executing tests on legacy devices (iPhone 6 @ iOS 10.x, iPhone 6 @ iOS 9.x)..."
+
+xcodebuild clean test \
+  -workspace "Stripe.xcworkspace" \
+  -scheme "StripeiOS" \
+  -configuration "Debug" \
+  -sdk "iphonesimulator" \
+  -destination "platform=iOS Simulator,name=iPhone 6,OS=10.3.1" \
+  -destination "platform=iOS Simulator,name=iPhone 6,OS=9.3" \
+  -skip-testing:"StripeiOS Tests/STPAddCardViewControllerLocalizationTests" \
+  -skip-testing:"StripeiOS Tests/STPPaymentMethodsViewControllerLocalizationTests" \
+  -skip-testing:"StripeiOS Tests/STPShippingAddressViewControllerLocalizationTests" \
+  -skip-testing:"StripeiOS Tests/STPShippingMethodsViewControllerLocalizationTests" \
+  -skip-testing:"StripeiOS Tests/STPImageLibraryTest" \
+  | xcpretty
+
+exit_code="${PIPESTATUS[0]}"
+
+if [[ "${exit_code}" != 0 ]]; then
+  die "xcodebuild exited with non-zero status code: ${exit_code}"
+fi
+
+# Execute tests on 32-bit device (iPhone 4s @ iOS 9.x)
+# - Skips snapshot tests because they're recorded for the iPhone 6 on the newest iOS version only
+# - Not sure why tests STPImageLibraryTest fail on older iOS versions
+info "Executing tests on 32-bit device  (iPhone 4s @ iOS 9.x)..."
+
+xcodebuild clean test \
+  -workspace "Stripe.xcworkspace" \
+  -scheme "StripeiOS" \
+  -configuration "Debug" \
+  -sdk "iphonesimulator" \
+  -destination "platform=iOS Simulator,name=iPhone 4s,OS=9.3" \
```

Comment:
What happens when you try to stick this in the previous `xcodebuild` command? Does it fail because `Debug` only builds for the active architecture?

Did you consider DRYing this up instead of duplicating so many parameters?

Reply:
> What happens when you try to stick this in the previous xcodebuild command? Does it fail because Debug only builds for the active architecture?

Oops, forgot to leave a reason comment but yeah that's why. Never figured it out but your guess seems reasonable! The code duplication is definitely annoying. I haven't tried parameters in a variable before but I'll look into it.

Reply:
Duplication doesn't look too bad after adding the `ONLY_ACTIVE_ARCH=NO` usage